leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/LwM2mQueue.java

```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+ * Copyright (c) 2017 RISE SICS AB.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     RISE SICS AB - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import org.eclipse.leshan.server.registration.Registration;
+import org.eclipse.leshan.util.Validate;
+
+/**
+ * Class that contains all the necessary elements to handle the queue mode. Every registration object that uses Queue
+ * Mode has a LwM2mQueue object for handling it.
+ */
+
+public class LwM2mQueue {
+
+    /* The state of the client: Awake or Sleeping */
+    private Presence state;
+
+    /* Elements to handle the time the client waits before going to sleep */
+    private Timer clientAwakeTimer;
+
+    private TimerTask clientAwakeTask;
+
+    private int clientAwakeTime;
+
+    /* Registration and Presence Service for notifying sleeping */
+    private Registration registration;
+
+    private PresenceService presenceService;
```

Comment:
We should not have a `PresenceService` attribute. This create a cyclic dependency which is often means bad design.